internal/clients/http/azure/create_vm.go

```diff
@@ -0,0 +1,395 @@
+		Location: to.Ptr(location),
+		Properties: &armnetwork.VirtualNetworkPropertiesFormat{
+			AddressSpace: &armnetwork.AddressSpace{
+				AddressPrefixes: []*string{
+					to.Ptr("10.42.0.0/16"),
+				},
+			},
+			//Subnets: []*armnetwork.Subnet{
+			//	{
+			//		Name: to.Ptr(subnetName+"3"),
+			//		Properties: &armnetwork.SubnetPropertiesFormat{
+			//			AddressPrefix: to.Ptr("10.1.0.0/24"),
+			//		},
+			//	},
+			//},
+		},
+	}
+
+	pollerResponse, err := vnetClient.BeginCreateOrUpdate(ctx, resourceGroupName, name, parameters, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := pollerResponse.PollUntilDone(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return &resp.VirtualNetwork, nil
+}
+
+func (c *client) createSubnets(ctx context.Context, resourceGroupName string, vnetName string, name string) (*armnetwork.Subnet, error) {
+	ctx, span := otel.Tracer(TraceName).Start(ctx, "createSubnets")
+	defer span.End()
+
+	subnetClient, err := c.newSubnetsClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	parameters := armnetwork.Subnet{
+		Properties: &armnetwork.SubnetPropertiesFormat{
+			AddressPrefix: to.Ptr("10.42.42.0/24"),
+		},
+	}
+
+	pollerResponse, err := subnetClient.BeginCreateOrUpdate(ctx, resourceGroupName, vnetName, name, parameters, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := pollerResponse.PollUntilDone(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return &resp.Subnet, nil
+}
+
+func (c *client) createPublicIP(ctx context.Context, location string, resourceGroupName string, name string) (*armnetwork.PublicIPAddress, error) {
+	ctx, span := otel.Tracer(TraceName).Start(ctx, "createPublicIP")
+	defer span.End()
+
+	publicIPAddressClient, err := c.newPublicIPAddressesClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	parameters := armnetwork.PublicIPAddress{
+		Location: to.Ptr(location),
+		Properties: &armnetwork.PublicIPAddressPropertiesFormat{
+			PublicIPAllocationMethod: to.Ptr(armnetwork.IPAllocationMethodStatic), // Static or Dynamic
+		},
+	}
+
+	pollerResponse, err := publicIPAddressClient.BeginCreateOrUpdate(ctx, resourceGroupName, name, parameters, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := pollerResponse.PollUntilDone(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+	return &resp.PublicIPAddress, err
+}
+
+func (c *client) createNetworkSecurityGroup(ctx context.Context, location string, resourceGroupName string, name string) (*armnetwork.SecurityGroup, error) {
+	ctx, span := otel.Tracer(TraceName).Start(ctx, "createNetworkSecurityGroup")
+	defer span.End()
+
+	nsgClient, err := c.newSecurityGroupsClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	parameters := armnetwork.SecurityGroup{
+		Location: to.Ptr(location),
+		Properties: &armnetwork.SecurityGroupPropertiesFormat{
+			SecurityRules: []*armnetwork.SecurityRule{
+				// Windows connection to virtual machine needs to open port 3389,RDP
+				// inbound
```

Comment:
Irrelevant comment I guess :-D

Reply:
Yeah, leftover from the sample cleanup :)